widget/entry_cursor_anim.go

```diff
@@ -0,0 +1,156 @@
+package widget
+
+import (
+	"image/color"
+	"sync"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/theme"
+)
+
+type safeCounter struct {
+	mu  sync.RWMutex
+	cnt int
+}
+
+func (c *safeCounter) Inc(n int) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.cnt += n
+}
+
+func (c *safeCounter) Value() int {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.cnt
+}
+
+func (c *safeCounter) Reset() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.cnt = 0
+}
+
+const cursorStopTimex10ms = 35 // stop time in multiple of 10 ms
+
+type cursorState int
+
+const (
+	cursorStateRunning cursorState = iota
+	cursorStateTemporarilyStopped
```

Comment:
It could be shorter as `cursorStatePaused` :)

Reply:
Yes, should be. However, it isn't really paused, it is stopped. I had that name before (paused) , but since the discussion about `TemporaryPause` function, I think it is better to call it `TemporarilyStopped` (to avoid confusions) because when the animation start again, it will not be restarted from the previous state, it will be restarted from the beginning.

Reply:
Fair point. perhaps "Interrupted" or something that is not Paused but also is not re-using Stopped?

Reply:
Yes, that's a good name, I will update!

Reply:
Should I also rename `TemporaryStop` to `Interrupt`?